src/client.ts

```diff
@@ -46,4 +46,40 @@ export class SlashauthClient {
       }
     );
   }
+
+  async validateToken({
+    token,
+  }: ValidateTokenArguments): Promise<ValidateTokenResponse> {
+    const resp =  this.apiClient.get<ValidateTokenAPIResponse>(
+      `/validate_token`,
+      {
+        queryParameters: {
+          params: {
+            client_id: this.client_id,
+            token,
+          },
+        },
+      }
+    );
+
+    if (!resp) {
+      throw new Error("token is not valid")
+    }
+
+    const encodedClaims = token.split('.')[1];
+
+    if (!encodedClaims) {
+      throw new SyntaxError('malformed token');
+    }
+
+    const decodedClaims = JSON.parse(atob(encodedClaims)) as DecodedToken;
+
+    return {
+      address: decodedClaims.sub,
```

Comment:
Let's add chain here too. For now let's just hardcode it to eth but it makes it easier to modify in the future

Reply:
why would we add chain here? this token won't be tied to a chain since the app could have requirements for different chains